test/Common/TestOrchestrations.cs

```diff
@@ -1549,5 +1549,32 @@ public static async Task<string> ReplaySafeLogger_OneLogMessage([OrchestrationTr
             string output = await ctx.CallActivityAsync<string>(nameof(TestActivities.Hello), input);
             return output;
         }
+
+        public static async Task<string> GetOrchestrationVersion([OrchestrationTrigger] IDurableOrchestrationContext ctx, ILogger log)
+        {
+            string orchVersion = VersionToString(ctx.Version);
+            string subOrchVersion = await ctx.CallSubOrchestratorAsync<string>(nameof(GetOrchestrationVersion_SubOrchestrator), null);
+            return $"Orchestration: {orchVersion}; Sub-orchestration: {subOrchVersion}";
+        }
+
+        public static async Task<string> GetOrchestrationVersion_AfterExternalEvent([OrchestrationTrigger] IDurableOrchestrationContext ctx, ILogger log)
+        {
+            ctx.SetCustomStatus("Waiting");
+            await ctx.WaitForExternalEvent<object>("Resume");
+
+            string orchVersion = VersionToString(ctx.Version);
+            string subOrchVersion = await ctx.CallSubOrchestratorAsync<string>(nameof(GetOrchestrationVersion_SubOrchestrator), null);
+            return $"Orchestration: {orchVersion}; Sub-orchestration: {subOrchVersion}";
+        }
+
+        public static async Task<string> GetOrchestrationVersion_SubOrchestrator([OrchestrationTrigger] IDurableOrchestrationContext ctx, ILogger log)
+        {
+            return await Task.FromResult(VersionToString(ctx.Version));
+        }
+
+        private static string VersionToString(string version)
+        {
+            return version == null ? "null" : $"'{version}'";

```

Comment:
Does this evaluate to 'version' or version? Just checking

Reply:
If `version` is not `null`, this code returns the version string surrounded by single quotes. I just wanted to be very explicit in the tests and clearly distinguish between `null`, an empty string, and a non-empty string that accidentally happens to contain `"null"`.

Technically, all of these work and lead to different results. I was thinking about introducing some validation there (perhaps prohibit either nulls, or empty strings, or both), but decided against that for now. `null` can be caused by `appVersion` not specified in `host.json` and an empty string can be caused by an env variable reference that evaluated to an empty string because the variable is missing. Both are potentially valid use cases, so I want to make sure they lead to reasonable behavior.